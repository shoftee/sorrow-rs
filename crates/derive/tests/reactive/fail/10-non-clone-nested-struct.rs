use sorrow_derive::Reactive;

#[derive(Reactive)]
struct OuterTest {
    #[reactive(nested)]
    nested: InnerTest,
}

#[derive(Reactive)]
struct InnerTest {
    test_field: i32,
}

fn main() {}
