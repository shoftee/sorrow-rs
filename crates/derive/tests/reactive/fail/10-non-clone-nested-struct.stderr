error[E0277]: the trait bound `InnerTest: Clone` is not satisfied
 --> tests/reactive/fail/10-non-clone-nested-struct.rs:3:10
  |
3 | #[derive(Reactive)]
  |          ^^^^^^^^ the trait `Clone` is not implemented for `InnerTest`
  |
  = note: this error originates in the derive macro `Reactive` (in Nightly builds, run with -Z macro-backtrace for more info)
help: consider annotating `InnerTest` with `#[derive(Clone)]`
  |
10+ #[derive(Clone)]
11| struct InnerTest {
  |
